x/btcstkconsumer/keeper/grpc_query.go

```diff
@@ -131,3 +132,53 @@ func (k Keeper) ConsumersRegistry(c context.Context, req *types.QueryConsumersRe
 	resp := &types.QueryConsumersRegistryResponse{ConsumerRegisters: consumersRegisters}
 	return resp, nil
 }
+
+func (k Keeper) ConsumerActive(goCtx context.Context,
+	req *types.QueryConsumerActiveRequest) (*types.QueryConsumerActiveResponse,
+	error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	if req.ConsumerId == "" {
+		return nil, status.Error(codes.InvalidArgument, "consumer id cannot be empty")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	consumer, err := k.GetConsumerRegister(ctx, req.ConsumerId)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "consumer cannot be found")
+	}
+
+	var active bool
+	switch consumer.Type() {
+	case types.ConsumerType_COSMOS:
+		channelID := consumer.GetCosmosConsumerMetadata().ChannelId
+		active = k.channelKeeper.ConsumerHasIBCChannelOpen(ctx, req.ConsumerId, channelID)
+	case types.ConsumerType_ROLLUP:
+		address := consumer.GetRollupConsumerMetadata().FinalityContractAddress
+		contractAddress, err := sdk.AccAddressFromBech32(address)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, "invalid contract address")
+		}
+
+		queryMsg := []byte(`{"config":{}}`)
+		queryRes, err := k.wasmKeeper.QuerySmart(goCtx, contractAddress,
```

Comment:
I think we do not need to use `QuerySmart` smart here and it is enough if we use:
```go
GetContractInfo(ctx context.Context, contractAddress sdk.AccAddress) *wasmtypes.ContractInfo
```
as `ContractInfo` struct contrains:
```go
// Label is optional metadata to be stored with a contract instance.
Label string `protobuf:"bytes,4,opt,name=label,proto3" json:"label,omitempty"`
```
label field.

What we discussed with @SebastianElvis is that contract should be considered inactive if the `Label != ""` i.e if label is different than empty string.